Automattic/jetpack
# Restaurant Menu CPT: Replace most jQuery with Javascript

## Proposed changes:

* While beginning the project to move Restaurant Menu CPT to the Classic Theme Helper package, the linter noticed a large number of issues with the three jQuery files. This PR converts two of those files from the original module location to Javascript.
* The intention here is to add this change before copying the changed code over (so the changes will be live before the CPT is required from the Classic Theme Helper package, to ensure no additional issues later).
* The addition of the mutation observer in `many-items.js` is primarily to help prevent memory leaks by cleaning up event listeners, as well as being more efficient (for example by avoiding polling).

### Other information:


## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

The testing process is the same for self-hosted, WoA and Simple (with Simple, apply the change using the command in the generated comment below, for a sandboxed site).

Test original behavior:
* To test the Restaurant Menus CPT, you'll need a theme that has support for this (eg Confit). 
* On trunk / latest stable of Jetpack, add several new menu items from the new Food Menu in the admin menu. 
* Add some new sections, and add menu items to specific sections (see the Menu Sections area in the menu editor). Ensure some sections are parents to other sections
* Attempt adding multiple menu items at once using the 'Add Many Items' menu option.
* Using quick edit from the main menu list for specific item, change the sections they belong to in some cases.
* Try dragging and dropping menu items to different positions and sections. 
* Click save new order, everything should save correctly.

Test new behavior:
* Apply this PR using the beta tester plugin, or apply in a local dev environment, and follow the same steps as above. Everything should work as it did before, with no console errors showing.
